libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSource.kt

```diff
@@ -348,6 +323,51 @@ class MapboxNavigationViewportDataSource(
      * @see [getViewportData]
      */
     fun evaluate() {
+        updateFollowingData()
+        updateOverviewData()
+
+        options.followingFrameOptions.run {
+            followingCameraOptions =
+                CameraOptions.Builder().apply {
+                    if (centerUpdatesAllowed) {
+                        center(followingCenterProperty.get())
+                    }
+                    if (zoomUpdatesAllowed) {
+                        zoom(followingZoomProperty.get())
+                    }
+                    if (bearingUpdatesAllowed) {
+                        bearing(followingBearingProperty.get())
+                    }
+                    if (pitchUpdatesAllowed) {
+                        pitch(followingPitchProperty.get())
+                    }
+                    if (paddingUpdatesAllowed) {
+                        padding(appliedFollowingPadding)
+                    }
+                }.build()
+        }
+
+        options.overviewFrameOptions.run {
+            overviewCameraOptions =
+                CameraOptions.Builder().apply {
+                    if (centerUpdatesAllowed) {
+                        center(overviewCenterProperty.get())
+                    }
+                    if (zoomUpdatesAllowed) {
+                        zoom(overviewZoomProperty.get())
+                    }
+                    if (bearingUpdatesAllowed) {
+                        bearing(overviewBearingProperty.get())
+                    }
+                    if (pitchUpdatesAllowed) {
+                        pitch(overviewPitchProperty.get())
+                    }
+                    if (paddingUpdatesAllowed) {
+                        padding(overviewPadding)
+                    }
+                }.build()
+        }
```

Comment:
Could we extract each of these blocks into `private` `fun`ctions?

Reply:
It was private fun before but I find the current approach more readable.